python/src/aiconfig/editor/client/src/components/JSONEditor.tsx

```diff
@@ -0,0 +1,78 @@
+import { createStyles, useMantineTheme } from "@mantine/core";
+import { Editor, Monaco } from "@monaco-editor/react";
+import { JSONObject, JSONValue } from "aiconfig";
+import { memo } from "react";
+
+type Props = {
+  content: JSONValue;
+  onChangeContent: (value: JSONValue) => void;
+  schema?: JSONObject;
+};
+
+const useStyles = createStyles(() => ({
+  monacoEditor: {
+    minHeight: "300px",
+  },
+}));
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function configureEditor(
+  _editor: Monaco["editor"]["IStandaloneCodeEditor"],
+  monaco: Monaco,
+  schema?: JSONObject
+) {
+  // Validate the text against PromptInput schema
+  monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+    validate: true,
+    schemas: [
+      {
+        uri: "https://json.schemastore.org/aiconfig-1.0",
```

Comment:
How do we have this stored here?

Reply:
Sarmad committed it there (https://github.com/SchemaStore/schemastore/pull/3474)